app/hooks/use-media-query.tsx

```diff
@@ -0,0 +1,44 @@
+import {useEffect, useState} from 'react';
+
+function useMediaQuery(query: string): boolean {
+  const getMatches = (query: string): boolean => {
+    // Prevents SSR issues
+    if (typeof window !== 'undefined') {
+      return window.matchMedia(query).matches;
+    }
+    return false;
+  };
+
+  const [matches, setMatches] = useState<boolean>(false);
+
+  function handleChange() {
+    setMatches(getMatches(query));
+  }
+
+  useEffect(() => {
+    const matchMedia = window.matchMedia(query);
+
+    // Triggered at the first client-side load and if query changes
+    handleChange();
+
+    // Listen matchMedia
+    if (matchMedia.addListener) {
+      matchMedia.addListener(handleChange);
+    } else {
+      matchMedia.addEventListener('change', handleChange);
+    }
+
+    return () => {
+      if (matchMedia.removeListener) {
+        matchMedia.removeListener(handleChange);
+      } else {
+        matchMedia.removeEventListener('change', handleChange);
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
? not sure if i get this!

Reply:
This comment removes a warning about `useEffect` dependencies array since `handleChange` is used inside of it. Although I can remove the comment, add `handleChange` to the dependency array, and make `handleChange` a `useCallback` function.

Reply:
Yes, please do Saul, this type of eslint disable just snowball in the future!